File: react/src/api/models/acroStack/services/dtos/cms/CreatePageInput.ts
/* oxlint-disable */

/**
 * @type object
 */
export type AcroStackServicesDtosCmsCreatePageInput = {
  /**
   * @minLength 0
   * @maxLength 256
   * @type string
   */
  title: string;
  /**
   * @minLength 0
   * @maxLength 128
   * @type string
   */
  slug: string;
  /**
   * @minLength 1
   * @type string
   */
  content: string;
  /**
   * @minLength 0
   * @maxLength 512
   * @type string | undefined
   */
  description?: string | null;
};

File: AcroStack/Services/Dtos/Cms/CreatePageInput.cs
using System.ComponentModel.DataAnnotations;

namespace AcroStack.Services.Dtos.Cms;

/// <summary>Input for creating a new <see cref="PageDto"/>.</summary>
public class CreatePageInput
{
    [Required]
    [StringLength(256)]
    public string Title { get; set; } = string.Empty;

    [Required]
    [StringLength(128)]
    public string Slug { get; set; } = string.Empty;

    [Required]
    public string Content { get; set; } = string.Empty;

    [StringLength(512)]
    public string? Description { get; set; }
}
